weave-js/src/components/PagePanelComponents/Home/Browse3/feedback/HumanFeedback/tsHumanFeedback.ts

```diff
@@ -0,0 +1,58 @@
+export type FeedbackTypeParts = {
+  field: string;
+  userDefinedType: string;
+  feedbackType: string;
+  displayName: string;
+};
+
+export const parseFeedbackType = (
+  inputField: string
+): FeedbackTypeParts | null => {
+  // input: summary.weave.feedback.wandb.annotation.Numerical-field-2.payload.value
+  // or input: wandb.annotation.Numerical-field-2.payload.value
+  //
+  // output:
+  // field: wandb.annotation.Numerical-field-2
+  // userDefinedType: Numerical-field-2
+  // type: annotation
+  // displayName: Annotation.Numerical-field-2
+
+  // If the field is coming from the flattened table, remove the
+  // summary portion
+  const field = inputField.startsWith('summary.weave.feedback')
+    ? inputField.replace('summary.weave.feedback.', '')
+    : inputField;
+  const deBracketed = field.replace(/\[.*\]/g, '');
```

Comment:
the fact that this is looking for a bracket does not make sense to me given the comment above

Reply:
i think this is just a no-op right now

Reply:
ill update the comment, but I am using this fn in one place where we go from backend filter -> displayName 

```javascript
export const getFieldLabel = (field: string): string => {
  if (field.startsWith('feedback.')) {
    // Here the field is coming from convertFeedbackFieldToBackendFilter
    // so the field should start with 'feedback.' if feedback
    const parsed = parseFeedbackType(field);
    if (parsed === null) {
      return field;
    }
    return parsed.displayName;
  }
  return FIELD_LABELS[field] ?? field;
};
```

Reply:
hmmm, i would probably split those, but sure. It seems like a sufficiently different case

Reply:
hmmm, i would probably split those, but sure. It seems like a sufficiently different case